src/controllers/auth.controller.ts

```diff
@@ -0,0 +1,108 @@
+import { Router } from "express";
+import { UserService } from "../services/user/user.service";
+import jwt from "jsonwebtoken";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const router = Router();
+const userService = new UserService();
+
+// User Registration
+router.post("/signup", async (req, res) => {
+  const { username, email, password } = req.body;
+
+  try {
+    const savedUser = await userService.createUser(username, email, password);
```

Comment:
Add check for uniqueness for email and username. Username and email must be unique